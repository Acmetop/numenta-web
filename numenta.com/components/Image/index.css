/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.mw {
  composes: mw-100 from 'tachyons';
}
.w {
  composes: w-100 from 'tachyons';
}


.center {
  composes: center from 'tachyons';
}
.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}
.round {
  composes: br-100 from 'tachyons';
}
.shadow {
  composes: shadow-4 from 'tachyons';
}


.image {
  composes: br2 from 'tachyons';
  composes: mb1 from 'tachyons';
}
