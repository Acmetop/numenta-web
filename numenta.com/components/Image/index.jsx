import classNames from 'classnames'
import {prefixLink} from 'gatsby-helpers'  // eslint-disable-line import/no-unresolved, max-len
import React from 'react'

import styles from './index.css'


const Image = ({alt, border, onClick, respond, shadow, src, title}) => {
  const classes = [styles.image]
  const prefix = prefixLink()
  let source = src

  if (prefix && source && !source.match(prefix)) {
    source = prefixLink(source)
  }

  if (border) classes.push(styles.border)
  if (respond) classes.push(styles[respond])
  if (shadow) classes.push(styles.shadow)

  return (
    <img
      alt={alt}
      className={classNames(...classes)}
      onClick={onClick}
      src={source}
      title={title}
    />
  )
}

Image.propTypes = {
  alt: React.PropTypes.string.isRequired,
  border: React.PropTypes.bool,
  onClick: React.PropTypes.func,
  respond: React.PropTypes.oneOf(['mw', 'w']),
  shadow: React.PropTypes.bool,
  src: React.PropTypes.string.isRequired,
  title: React.PropTypes.string,
}

Image.defaultProps = {
  border: false,
  shadow: true,
}

export default Image
