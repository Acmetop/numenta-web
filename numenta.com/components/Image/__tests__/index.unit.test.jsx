import React from 'react'
import renderer from 'react-test-renderer'

import Image from '../../Image'


describe('<Image /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Image />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
