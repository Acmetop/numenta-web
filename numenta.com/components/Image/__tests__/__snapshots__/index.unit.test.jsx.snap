exports[`<Image /> React component unit test suite Renders correctly 1`] = `
<img
  alt={undefined}
  className="image shadow"
  onClick={undefined}
  src={undefined}
  title={undefined} />
`;
